src/Aspire.Hosting.Milvus/MilvusBuilderExtensions.cs

```diff
@@ -190,4 +259,44 @@ private static void ConfigureAttuContainer(EnvironmentCallbackContext context, M
         // This will need to be refactored once updated service discovery APIs are available
         context.EnvironmentVariables.Add("MILVUS_URL", $"{resource.PrimaryEndpoint.Scheme}://{resource.Name}:{resource.PrimaryEndpoint.TargetPort}");
     }
+    internal static MilvusClient CreateMilvusClient(IServiceProvider sp, string? connectionString)
+    {
+        if (connectionString is null)
```

Comment:
Ideally we would share this logic: https://github.com/dotnet/aspire/blob/65c69024e5ebfe0c7914e6fbcbfea12aa359bd71/src/Components/Aspire.Milvus.Client/MilvusClientSettings.cs#L11

Reply:
I agree with @davidfowl we should do this because the created Milvus connection string in app-host isn't compatible with MilvusClient and those properties (database, endpoint, key) should be extracted before creating MilvusClient.

Reply:
Was there a reason this wasn't done?